src/main/java/core/basesyntax/service/FileReader.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.service.TransactionParser.parseTransaction;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class FileReader {
+    private TransactionProcessor transactionProcessor;
+
+    private FileReader(TransactionProcessor transactionProcessor) {
+        this.transactionProcessor = transactionProcessor;
+    }
+
+    public void read(String fileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new java.io.FileReader(fileName))) {
+            bufferedReader.readLine(); //reading first line because it contains fields names
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                transactionProcessor.processTransaction(parseTransaction(line));

```

Comment:
Please follow single responsibility principle.
FileReader should read file, it shouldn't call parser or processor.
Make calls separately in main method